libs/components/src/progress/progress.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input } from "@angular/core";
+
+type SizeTypes = "small" | "default" | "large";
+
+const SizeClasses: Record<SizeTypes, string[]> = {
+  small: ["tw-h-1"],
+  default: ["tw-h-4"],
+  large: ["tw-h-6"],
+};
+
+@Component({
+  selector: "bit-progress",
+  templateUrl: "./progress.component.html",
+})
+export class ProgressComponent {
+  @Input() barWidth = 0;
+  @Input() showText = true;
+  @Input() size: SizeTypes = "default";
+  @Input() text: string;
```

Comment:
Please mark this as optional (for when we turn on strict mode).

```suggestion
  @Input() text?: string;
```